core/state/statedb.go

```diff
@@ -977,7 +1061,10 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
 // - Add the contents of the optional tx access list (2930)
```

Comment:
They are not getters or setters that use an address to get or set the state. It's unlikely that an out-of-context address would make it into the temporary state for it to be committed to the database. However, since I can't rule it out, I've added checks.

Reply:
Added the checks and lots of error handling